core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -638,6 +652,43 @@ static Parameter ofRequestObject(Class<?> type,
             this.name = name;
             this.defaultValue = defaultValue;
             this.requestConverterFunction = requestConverterFunction;
+
+            if (type.isEnum()) {
+                final Set<? extends Enum<?>> enumInstances = EnumSet.allOf(type.asSubclass(Enum.class));
+                Map<String, ? extends Enum<?>> upperCaseEnumMap = enumInstances.stream().collect(
+                        toImmutableMap(e -> Ascii.toUpperCase(e.name()), Function.identity(), (e1, e2) -> e1));
+                if (enumInstances.size() != upperCaseEnumMap.size()) {
+                    upperCaseEnum = null;
+                    isCaseSensitiveEnum = true;
+                } else {
+                    upperCaseEnum = upperCaseEnumMap;
+                    isCaseSensitiveEnum = false;
+                }
+            } else {
+                upperCaseEnum = null;
+                isCaseSensitiveEnum = false;
+            }
+        }
+
+        <T> T getEnumElement(String str) {
+            final Object o;
+            if (isCaseSensitiveEnum) {
+                o = Enum.valueOf(type.asSubclass(Enum.class), str);
+            } else {
+                o = upperCaseEnum.get(Ascii.toUpperCase(str));
```

Comment:
With the comment above addressed, this could be simplified:

    o = enumMap.get(isCaseSensitiveEnum ? str : Ascii.toLowerCase(str));

Also, please note you did not handle the `IllegalArgumentException` raised by `Enum.valueOf()`, although `valueOf()` call should be gone after addressing the review comments.

Reply:
Checked & Updated Thanks!!